### ISSUE TITLE
What is the coordinate convention for the saved dataparser_transforms.json ?

### ISSUE THREAD

**Issue by AruniRC** [nerfstudio-project/nerfstudio#3484]
It is not clear if the `dataparser_transforms.json` "transform" matrix is in OpenCV or OpenGL convention, if the data pre-processing, say, was done using COLMAP (which would be using OpenCV).

This makes it hard to correctly apply this transformation, if I am trying to convert camera poses in the input coordinate system (ie. whatever COLMAP gives) into the coordinate system used by the trained Nerf.

**Comment by lyg1597** [comment]
Hope this answer helps with the relationship between colmap coordinate frame and trained nerf coordinate frame. 
https://github.com/nerfstudio-project/nerfstudio/issues/3488#issuecomment-2430013120

**Comment by AruniRC** [comment]
Thank you for the response @lyg1597 .  

However, I think one more bit of information is needed to correctly apply the transform you pointed out: we still do not know the coordinate convention of the transformation matrix and scale factor can be found in `dataparser_transforms.json`. That is, is it saved in the OpenCV format (which colmap uses) or the OpenGL format (which nerfstudio uses).

**Comment by lyg1597** [comment]
I am not that familar with the difference/conversion between OpenCV and OpenGL convention, but I can describe what I am doing. The steps that I am using is:
1. Collecct images and camera poses from gazebo simulator. 
2. Use script from https://github.com/nerfstudio-project/nerfstudio/issues/3101#issuecomment-2130324279 (thanks for your solution) to perform the first coordinate transformation.
3. Apply the second coordinate transformation I described in the previous comment https://github.com/nerfstudio-project/nerfstudio/issues/3488#issuecomment-2430013120.
4. Render images at the resulting camera pose.
The images that I am getting from the last step looks exactly (except nerf artifect) the same as the images directly from the simulator. Hope this is helpful for your problem.

**Comment by AruniRC** [comment]
@lyg1597 - this is super helpful, thank you for the details. The step 4 that you mention is basically calling `ns-render` from the cmd line on the `camera_path.json` you generated in step 3?

**Comment by lyg1597** [comment]
No, I am using my own [renderer](https://github.com/lyg1597/NerfExp/blob/main/gazebo_nerf_exp/ns_renderer.py) because I want minimum code that can give me single images. But all functions I am using is from nerfstudio/ns-render.
